src/core/session.cpp

```diff
@@ -60,6 +77,31 @@ class standard_logger_impl : public logger_impl
         return lastQuery_;
     }
 
+    virtual std::string get_last_query_with_context() const
+    {
+        if (queryParams_.empty()) {
```

Comment:
Minor but this doesn't conform to SOCI coding style:

```suggestion
        if (queryParams_.empty())
        {
```